imageroot/bin/discover-service

```diff
@@ -14,17 +14,39 @@ import sys
 import json
 import agent
 
-mail_server = os.environ["MAIL_SERVER"]
 rdb = agent.redis_connect(host="127.0.0.1") # full read-only access on every key
 
-# we query about TCP service provided by the mail server
-smtp_port = rdb.hget(f"module/{mail_server}/srv/tcp/submission", "port") or ""
-imap_port = rdb.hget(f"module/{mail_server}/srv/tcp/imap", "port") or ""
-# we query about the IP in the cluster of the mail server
-smtp_server = rdb.hget(f"module/{mail_server}/srv/tcp/submission", "host") or ""
-imap_server = rdb.hget(f"module/{mail_server}/srv/tcp/imap", "host") or ""
 
-envfile = "roundcubemail_default_port.env"
+imap = agent.list_service_providers(rdb, 'imap', 'tcp', {
+    'module_id': os.getenv('MAIL_SERVER', '')
+})
+
+if not imap:
+    imap = agent.list_service_providers(rdb, 'imap', 'tcp', {
+        'module_uuid': os.getenv('MAIL_MODULE_UUID', '')
+    })
+
+
+smtp = agent.list_service_providers(rdb, 'submission', 'tcp', {
+    'module_id': os.getenv('MAIL_SERVER', '')
+})
+
+if not smtp:
+    smtp = agent.list_service_providers(rdb, 'submission', 'tcp', {
+        'module_uuid': os.getenv('MAIL_MODULE_UUID', '')
+    })
+
+if imap:
+    imap_port = imap[0]['port']
+    imap_server = imap[0]['host']
+    user_domain = imap[0]['user_domain']
+
+if smtp:
+    smtp_port = smtp[0]['port']
+    smtp_server = smtp[0]['host']
+
```

Comment:
If the services cannot be found we can exit with an error, so roundcube is not started at all.

```
    user_domain = imap[0]['user_domain']
else:
    # exit with error

if smtp:
    smtp_port = smtp[0]['port']
    smtp_server = smtp[0]['host']
else:
    # exit with error
```